pkg/query/storeset.go

```diff
@@ -418,7 +438,14 @@ func (s *StoreSet) getHealthyStores(ctx context.Context, stores map[string]*stor
 					level.Warn(s.logger).Log("msg", "update of store node failed", "err", errors.Wrap(err, "dialing connection"), "address", addr)
 					return
 				}
-				st = &storeRef{StoreClient: storepb.NewStoreClient(conn), cc: conn, addr: addr, logger: s.logger}
+
+				var rule storepb.RulesClient
+
+				if _, ok := ruleAddrSet[addr]; ok {
```

Comment:
What's confusing here, is that if someone put something in `--rules` but not `--stores` it will be silently dropped

Reply:
Indeed, that is a fair edge case. I will implement a small assertion bailing out if `--rules` is given but not `--store`.

Reply:
I believe we cannot handle this at startup time because of the other discoverability flags, so instead logging out a warning as per https://github.com/thanos-io/thanos/pull/2240/commits/c054ca8dc8bff6e7632c5c0239d1dea212a8ca99

Reply:
:+1: